app/src/main/java/app/attestation/auditor/AttestationProtocol.java

```diff
@@ -743,151 +749,143 @@ private static Verified verifyStateless(final Certificate[] certificates,
                     throw new GeneralSecurityException("attest key challenge does not match");
                 }
 
-                if (!attestation1.getSoftwareEnforced().getAttestationApplicationId().equals(attestationApplicationId)) {
+                if (!attestation1.softwareEnforced.attestationApplicationId.orElseThrow(() ->
+                        new GeneralSecurityException("missing attest key application")).equals(attestationApplicationId)) {
                     throw new GeneralSecurityException("attest key application does not match");
                 }
 
                 // version sanity checks
-                if (attestation1.getAttestationVersion() != attestation.getAttestationVersion()) {
+                if (attestation1.attestationVersion != attestation.attestationVersion) {
                     throw new GeneralSecurityException("attest key attestation version does not match");
                 }
-                if (attestation1.getKeymasterVersion() != attestation.getKeymasterVersion()) {
+                if (attestation1.keymasterVersion != attestation.keymasterVersion) {
                     throw new GeneralSecurityException("attest key keymaster version does not match");
                 }
 
                 // OS version sanity checks
-                if (!teeEnforced1.getOsVersion().equals(teeEnforced.getOsVersion())) {
+                if (!teeEnforced1.osVersion.equals(teeEnforced.osVersion)) {
                     throw new GeneralSecurityException("attest key OS version does not match");
                 }
-                if (!teeEnforced1.getOsPatchLevel().equals(teeEnforced.getOsPatchLevel())) {
+                if (!teeEnforced1.osPatchLevel.equals(teeEnforced.osPatchLevel)) {
                     throw new GeneralSecurityException("attest key OS patch level does not match");
                 }
-                if (!teeEnforced1.getVendorPatchLevel().equals(teeEnforced.getVendorPatchLevel())) {
+                if (!teeEnforced1.vendorPatchLevel.equals(teeEnforced.vendorPatchLevel)) {
                     throw new GeneralSecurityException("attest key vendor patch level does not match");
                 }
-                if (!teeEnforced1.getBootPatchLevel().equals(teeEnforced.getBootPatchLevel())) {
+                if (!teeEnforced1.bootPatchLevel.equals(teeEnforced.bootPatchLevel)) {
                     throw new GeneralSecurityException("attest key boot patch level does not match");
                 }
 
-                if (!Arrays.equals(rootOfTrust1.getVerifiedBootHash(), rootOfTrust.getVerifiedBootHash())) {
+                if (!Arrays.equals(rootOfTrust1.verifiedBootHash.orElse(new byte[0]), rootOfTrust.verifiedBootHash.orElse(new byte[0]))) {
                     throw new GeneralSecurityException("attest key verified boot hash does not match");
                 }
             }
 
             attestKey = true;
-        } catch (final Attestation.KeyDescriptionMissingException e) {}
```

Comment:
We added Attestation.KeyDescriptionMissingException to the previous library and it needs to be added to the new library to reliably detect whether the attestation extension is present.